File: src/main/java/de/tarent/akka/java/remote/server/PongActor.java
package de.tarent.akka.java.remote.server;

import akka.actor.Props;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import de.tarent.akka.java.remote.model.Ping;
import de.tarent.akka.java.remote.model.Pong;

class PongActor extends UntypedActor {
    private final LoggingAdapter log = Logging.getLogger(getContext().system(), this);

    @Override
    public void onReceive(Object message) throws Throwable {
        if(Ping.match(message)) {
            log.info("Receiving a ping. Will pong back.");
            sender().tell(new Pong(), self());
        } else {
            unhandled(message);
        }

        Thread.sleep(500);
    }

    public static Props props() {
        return Props.create(PongActor.class);
    }
}

File: src/main/scala/de/tarent/scala/pingpong/PongActor.scala
package de.tarent.scala.pingpong

import akka.actor.{Actor, Props}
import akka.event.Logging

class PongActor extends Actor {
  val log = Logging(context.system, this)

  override def receive: Receive = {
    case Ping =>
      log.info("Receiving a PING! Will pong back.")
      sender ! Pong
  }
}

object PongActor {
  def props: Props = Props[PongActor]
}